src/main/java/com/google/api/gax/testing/MockStreamObserver.java

```diff
@@ -54,6 +54,7 @@ public void onNext(T message) {
   @Override
   public void onError(Throwable t) {
     errors.add(t);
+    future.set(actualMessages);
```

Comment:
Throwing an exception actually makes the surface behave like it is supposed to (like the real service will).